fava/plugins/link_statements.py

```diff
@@ -0,0 +1,68 @@
+"""Plugin that looks through all transactions and their postings for the
+`statement`-metadata-key, and if found it looks for corresponding Document
+entries that were added by beancount automatically through file discovery.
+
+It then adds a link to the transaction and the Document, and the tag
+"#statement" to the Document.
+"""
+import collections
+from os.path import join, dirname, normpath
+from beancount.core import data
+from beancount.core.compare import hash_entry
+
+StatementDocumentError = collections.namedtuple('StatementDocumentError',
+                                                'source message entry')
+
+__plugins__ = ['link_statements']
+
+
+def link_statements(entries, options_map):
+    errors = []
+
+    if 'documents' not in options_map or len(options_map['documents']) == 0:
+        return entries, errors
+
+    all_documents = [(i, entry) for i, entry in enumerate(entries)
+                     if isinstance(entry, data.Document)]
+
+    all_transactions = [(i, entry) for i, entry in enumerate(entries)
+                        if isinstance(entry, data.Transaction)]
+
+    for i, entry in all_transactions:
+        statements = [value for key, value in entry.meta.items()
+                      if key.startswith('statement')]
+
+        for posting in entry.postings:
+            if posting.meta:
+                statements.extend([value for key, value in posting.meta.items()
+                                   if key.startswith('statement')])
+
+        _hash = hash_entry(entry)
```

Comment:
Maybe the full hash is a bit much for the link. How about just the first 8 or 10 characters?